rtrlib/rtr/packets.c

```diff
@@ -947,6 +981,38 @@ static int rtr_store_router_key_pdu(struct rtr_socket *rtr_socket, const void *p
 	return RTR_SUCCESS;
 }
 
+/*
+ * @brief Appends the ASPA pdu to ary.
+ *
+ * @return RTR_SUCCESS On success
+ * @return RTR_ERROR On realloc failure
+ * @attention ary is not freed in this case, because it might contain data that is still needed
+ */
+static int rtr_store_aspa_pdu(struct rtr_socket *rtr_socket, const void *pdu, const unsigned int pdu_size,
+					struct pdu_aspa **ary, unsigned int *ind, unsigned int *size)
+{
+	assert(rtr_get_pdu_type(pdu) == ASPA);
+
+	if ((*ind) >= *size) {
+		*size += TEMPORARY_PDU_STORE_INCREMENT_VALUE;
+		void *tmp = lrtr_realloc(*ary, *size * pdu_size);
+
+		if (!tmp) {
+			const char txt[] = "Realloc failed";
+
+			RTR_DBG("%s", txt);
```

Comment:
`RTR_DBG("Realloc failed")` ?

Reply:
even worse, they're using that inside a message transmitted over the network. I'd say you absolutely should not send these internal error messages inside a protocol pdu...